### PR TITLE
PRO-3055 adds callback handler for proofs

### PR DESCRIPTION
**Pull request by matt-raffel-kiva**
Signed-off-by: Matt Raffel <mattr@kiva.org>

1 Adding callbacks for other messages
2 added id to callback registration, helpful for debugging

### CODE REVIEW

**Review by jsaur — DISMISSED** [review]
Looks good (one possible extra log message)

File: src/controller/agent.governance.ts [hunk 1]
```diff
@@ -113,22 +119,43 @@ export class AgentGovernance {
...
+        let result: boolean = false;
         // tslint:disable-next-line:forin
         for (const index in this.callbacks) {
             const registration: Registration = this.callbacks[index];
-            Logger.debug(`callback found ${registration.topic}`, registration);
+            Logger.debug(`callback found topic: ${registration.topic} id: ${registration.id}`);
             if (registration.topic === topic) {
-                await registration.func(agentUrl, agentId, adminApiKey, route, topic, body, token);
+                const funResult: boolean = await registration.func(agentUrl, agentId, adminApiKey, route, topic, body, token);
+                if (funResult === true) {
```
Comment (jeffk-kiva) [inline src/controller/agent.governance.ts:150]: Another option to reduce nesting would be to change this `if` block to `result = funResult || result`. Definitely not necessary though.

File: src/controller/handler/connections.ts [hunk 2]
```diff
@@ -69,6 +69,11 @@ export class Connections extends BaseAgentResponseHandler {
             throw new ProtocolException(ProtocolErrorCode.AGENCY_GOVERNANCE,`${route}/${topic} is not valid.`);
         }
 
+        // allow consumers to process credentials prior to and differently than provided by this handler.  If the callback
+        // returns true (explicitly), it means this handlers code should not be executed.
+        if (true === await this.agentGovernance.invokeHandler(agentUrl, agentId, adminApiKey, route, topic, body, token))
```
Comment (jeffk-kiva) [inline src/controller/handler/connections.ts:74]: Do you need the `true ===`? An if statement should automatically check the truthiness, I'd expect.

File: src/controller/handler/proof.ts [hunk 3]
```diff
@@ -68,6 +68,7 @@ export class Proofs extends BaseAgentResponseHandler {
             req.headers.Authorization = 'Bearer ' + token;
         }
         const res = await this.http.requestWithRetry(req);
+        Logger.info(`getCredentialsByReferentId results`, res.data);
```
[the remark this answers is not in the document]
Comment (matt-raffel-kiva) [reply src/controller/handler/proof.ts]: good catch. removed log statement.  updated

**Review by jsaur — APPROVED** [review]

**Review by jeffk-kiva — APPROVED** [review]

**Pull request merged**